src/components/application_manager/test/commands/hmi/activate_app_request_test.cc

```diff
@@ -0,0 +1,117 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "application_manager/commands/hmi/activate_app_request.h"
+#include "utils/shared_ptr.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/commands/command_impl.h"
+#include "commands/commands_test.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+namespace activate_app_request {
+
+using ::testing::_;
+namespace am = ::application_manager;
+namespace strings = ::application_manager::strings;
+using am::commands::MessageSharedPtr;
+using am::commands::ActivateAppRequest;
+using am::commands::CommandImpl;
+
+typedef ::utils::SharedPtr<ActivateAppRequest> ActivateAppRequestPtr;
+
+MATCHER_P(CheckMessage, level, "") {
+  return level ==
+         static_cast<mobile_apis::HMILevel::eType>(
+             (*arg)[strings::msg_params][strings::activate_app_hmi_level]
+                 .asInt());
+}
+
+namespace {
+const uint32_t kAppId = 1u;
+const uint32_t kCorrelationId = 2u;
+}  // namespace
+
+class ActivateAppRequestTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMsgParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params[strings::app_id] = kAppId;
+    msg_params[strings::correlation_id] = kCorrelationId;
+    (*msg)[strings::msg_params] = msg_params;
+    (*msg)[strings::params][strings::app_id] = kAppId;
+    (*msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*msg)[strings::app_id] = kAppId;
+    return msg;
+  }
+};
+
+TEST_F(ActivateAppRequestTest, Run_SUCCESS) {
+  MessageSharedPtr msg = CreateMsgParams();
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kAppId)).WillOnce(Return(mock_app));
+#ifdef ENABLE_LOG
+  (*msg)[strings::msg_params][strings::activate_app_hmi_level] =
+      mobile_apis::HMILevel::HMI_FULL;
+#endif
+  ActivateAppRequestPtr command(CreateCommand<ActivateAppRequest>(msg));
+
+  EXPECT_CALL(app_mngr_, set_application_id(kCorrelationId, kAppId));
+#ifdef ENABLE_LOG
```

Comment:
@okozlovlux in case if it is moved code, please create TODO and task to resolve it.